apps/prairielearn/src/pages/instructorAssessmentAccess/instructorAssessmentAccess.types.ts

```diff
@@ -0,0 +1,20 @@
+import { z } from 'zod';
+
+export const AssessmentAccessRulesSchema = z.object({
+  mode: z.string(),
+  uids: z.string(),
+  start_date: z.string(),
+  formatted_start_date: z.string().optional(),
+  end_date: z.string(),
+  formatted_end_date: z.string().optional(),
+  credit: z.string(),
+  time_limit: z.string(),
+  password: z.string(),
+  exam_uuid: z.string().nullable(),
+  pt_course_id: z.string().nullable(),
+  pt_course_name: z.string().nullable(),
+  pt_exam_id: z.string().nullable(),
+  pt_exam_name: z.string().nullable(),
+  active: z.string(),
```

Comment:
Ditto here; this should ideally be a boolean. In general, I'd expect all the types here (with the exception of `pt_*`) to match their types in the database.

Reply:
Updated this to be a boolean in [1dfef07](https://github.com/PrairieLearn/PrairieLearn/pull/9934/commits/1dfef07527ed0296ff3cdfa7c58d85fce2304b97). We are now passing back the entire row in the query. I added a schema for `assessmentAccessRules` to `db-types` and now using this for the types here.